packages/components/src/lib/MarkdownViewer.tsx

```diff
@@ -25,20 +30,47 @@ function rehypeInlineCodeProperty() {
   };
 }
 
+type CustomComponents = Components & {
+  squiggleplayground: React.ComponentType<{ children: string }>;
+  squiggleeditor: React.ComponentType<{ children: string }>;
+};
+
 type MarkdownViewerProps = {
   md: string;
   textSize: "sm" | "xs";
   textColor?: "prose-stone" | "prose-slate";
   className?: string;
   backgroundColor?: string;
 };
+
 export const MarkdownViewer: React.FC<MarkdownViewerProps> = ({
   md,
   className,
   textColor,
   textSize,
   backgroundColor = "bg-slate-50",
 }) => {
+  const CustomSquigglePlayground = ({ children }: { children: string }) => {
```

Comment:
Hmmm. Inlining React components in other components is considered quite bad for React performance.

OTOH, examples for `<ReactMarkdown>` allow inline functions in `components` object, so I'm not sure what's going on there — it depends on whether `react-markdown` interprets these as components (creates JSX nodes) or just functions (calls them directly).

I tried to track it down in its source code but got to `hast-util-to-jsx-runtime` internals and gave up, don't want to spend too much time on this.

Anyway, my advice is to rename this from `CustomSquigglePlayground` to `makeSquigglePlayground`, and similarly rename `CustomSquiggleEditor` below, so that there'd be no risk of using these with JSX tags.